### DISCUSSION TITLE
Excel is getting corrupted using stream writer in latest version

### DISCUSSION THREAD

**Discussion by ramky7273** [qax-os/excelize#1594] [General]
With the latest version(2.7.1)  of this, in some cases excel is getting corrupted i.e. Excel is generated, but upon opening it says - Excel is corrupted. Similar issue was reported here https://github.com/qax-os/excelize/issues/1519 , however it was fixed in latest release. Still we could see same issue in latest ? 

any idea's or has anyone experienced it ?

**Comment by xuri** [comment]
Thanks for your feedback. Please try to upgrade and using the master branch code, and you can use the Go module to lock the version before the new version is released.

**Reply by xuri** [reply]
Hi @sis6789, thanks for your feedback. Which version of the Go re you using? Could you provide more details? such as code and steps to reproduce the issue.

**Reply by xuri** [reply]
Thanks for your feedback. Could you follow the issue template, and show us a complete, simplify, standalone example program or reproducible demo? Note that, there are some incompatible changes in the Go 1.21.0 encoding/xml library, which will cause generate corrupted workbook, please try to using Go 1.20 or previous Go version, also reference the issue #1465, #1595 and #1603.

**Comment by sis6789** [comment]
with go 1.20.7 version, everything is OK.
but with go 1.21 version, everything is not OK.

**Reply by xuri** [reply]
Please reference the issue #1465, #1595, #1603 and [golang/go#61881](https://github.com/golang/go/issues/61881).

**Comment by sis6789** [comment]
Is there a progress for this problem(v1.21)?

**Reply by xuri** [reply]
I have vendoring Go 1.16.15's encoding/xml package in github.com/xuri/xml, I will use it in the Excelize instead of encoding/xml standard library tomorrow, after that, you can upgrade the Excelize library with the master branch code, and these changes will be released in the next Excelize version, if Go 1.21.1 not yet released.

**Comment by keshavSahu-27** [comment]
Hi @xuri, I am getting this exact same issue with excelize version (2.7.1) and go version (1.21.0). It worked fine before but then the issue came in. I tried isolating the functionality to reproduce the exact issue. And found the issue surfaces in the library itself.
The strange thing I noticed is I had a test case which validates the excel exported must contains the exact data which was provided. To my surprise excelize was able to read this corrupt excel.

**Reply by xuri** [reply]
Thanks for your feedback. Could you show us a complete, standalone example program or reproducible demo?

**Reply by xuri** [reply]
Thanks for your feedback. Could you show us a complete, standalone example program with the main function?

**Reply by keshavSahu-27** [reply]
Sure, Attaching the zip file which contains the basic program and it also contains the generated excel also. 
Please have a look and let me know if anything is required to further investigate @xuri.
[goexcelize.zip](https://github.com/qax-os/excelize/files/12420989/goexcelize.zip)

**Reply by keshavSahu-27** [reply]
Hi @xuri , I was working on Go version 1.21 may be due to which it is having issues. And, it is working fine with Go version 1.19. Sorry for the confusion caused. I will edit the parent comment, to reflect the same.

**Comment by sis6789** [comment]
**xuri wrote:**

**_from facebook:_** There are some incompatible changes in the Go 1.21.0 encoding/xml library. I have given feedback to the Go team and created a patch for it golang/go#61881, and the Go team has added golang/go#62051 to the Go 1.21.1 milestone. I suggest using the Go 1.20.7 and previous Go released version or waiting for the next Go minor releases.

**Reply by sanfrancrisko** [reply]
I was encountering this issue on Go `1.21.0` - just upgraded to `1.21.1` and have been able to successfully open the XLSX spreadsheet in Excel, LibreOffice and Google Sheets, whereas previously couldn't. 

Thanks @xuri , @sis6789

**Comment by longphee** [comment]
Is this problem fixed on Go 1.21.5？and is it necessary to upgrade excelize to v2.8.0 ? @xuri

**Reply by xuri** [reply]
This problem is fixed on [Go 1.21.1](https://go-review.googlesource.com/466295), you needn't upgrade the Excelize library.
